features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/slowframes/DefaultUISlownessMetricDispatcherTest.kt

```diff
@@ -0,0 +1,215 @@
+    ExtendWith(TestConfigurationExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DefaultUISlownessMetricDispatcherTest {
+
+    @StringForgery
+    lateinit var fakeViewId: String
+
+    @FloatForgery(min = 0f, max = 100f)
+    var fakeSamplingRate: Float = 0f
+
+    @Forgery
+    lateinit var fakeSlowFramesConfiguration: SlowFramesConfiguration
+
+    @Mock
+    private lateinit var internalLogger: InternalLogger
+
+    private lateinit var testedDispatcher: DefaultUISlownessMetricDispatcher
+
+    @BeforeEach
+    fun `set up`() {
+        testedDispatcher = DefaultUISlownessMetricDispatcher(
+            config = fakeSlowFramesConfiguration,
+            internalLogger = internalLogger,
+            samplingRate = fakeSamplingRate
+        )
+    }
+
+    @Test
+    fun `M increment slowFramesCount W incSlowFrame`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.incSlowFrame(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(slowFramesCount = 1, config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M increment freezeFramesCount W incFreezeFrame`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.incFreezeFrame(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(freezeFramesCount = 1, config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M increment ignoredFramesCount W incFreezeFrame`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.incIgnoredFrame(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger).logMetric(
+            argThat { invoke() == DefaultUISlownessMetricDispatcher.UI_SLOWNESS_MESSAGE },
+            eq(buildMetricAttributesMap(ignoredFramesCount = 1, config = fakeSlowFramesConfiguration)),
+            eq(fakeSamplingRate),
+            eq(null)
+        )
+    }
+
+    @Test
+    fun `M send telemetry only once W sendSlowFramesTelemetry`() {
+        // Given
+        testedDispatcher.onViewCreated(fakeViewId)
+
+        // When
+        testedDispatcher.sendMetric(fakeViewId)
+        testedDispatcher.sendMetric(fakeViewId)
+
+        // Then
+        verify(internalLogger, times(1)).logMetric(
```

Comment:
nit: `times(1)` is the default mode

```suggestion
        verify(internalLogger).logMetric(
```